lib/withStaticQuery.js

```diff
@@ -1,17 +1,48 @@
 import React from 'react';
 import getDisplayName from './getDisplayName';
+import { Meteor } from 'meteor/meteor';
+import DataHydrator from './DataHydrator.js';
 
 export default function withStaticQueryContainer(WrappedComponent) {
     /**
      * We use it like this so we can have naming inside React Dev Tools
      * This is a standard pattern in HOCs
      */
     class GrapherStaticQueryContainer extends React.Component {
-        state = {
-            isLoading: true,
-            error: null,
-            data: [],
-        };
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                isLoading: true,
+                error: null,
+                data: [],
+            };
+
+            const { query } = props;
+
+            // Check the SSR query store for data 
+            if(Meteor.isClient && window && window.grapherQueryStore) {
+                const data = DataHydrator.getQueryData(query);
```

Comment:
create 2 additional methods for this, there's too much code coupled in the constructor.

Reply:
what if the query gets an error ? how is this treated ?